sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -458,7 +463,10 @@ object RewriteCorrelatedScalarSubquery extends Rule[LogicalPlan] {
         sys.error(s"Unexpected operator in scalar subquery: $lp")
     }
 
-    val resultMap = evalPlan(plan)
+    val resultMap = evalPlan(plan).mapValues { _.transform {
+        case a: Alias => a.newInstance() // Assigns a new `ExprId`
```

Comment:
I don't quite understand this change. Will it break parent nodes that refer to this alias?

Reply:
Sorry, but this follow the old semantics w/o `transformUpWithNewOutput`. I'm checking it now.

Reply:
I've checked that we don't need this change because of the same reason with https://github.com/apache/spark/pull/29585#discussion_r490318779. So, I've reverted it in the latest commit.